File: Leet-Code 75/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> pair = new HashMap<>();
        int num = 0;

        for(int i=0;i<nums.length;i++){
            num = target - nums[i];
            if(pair.containsKey(num)) return new int[]{pair.get(num),i};
            pair.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: Leet-Code 75 II/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        num = 0

        for i in range(len(nums)):
            num = target - nums[i]
            if num in map: return [i,map[num]]
            map[nums[i]] = i

        return [-1,-1]